apps/src/code-studio/pd/workshop_dashboard/components/session_form_part.jsx

```diff
@@ -79,30 +86,60 @@ var SessionFormPart = React.createClass({
     return null;
   },
 
+  // Return the time 1 minute before specified maxTime, bounded by MAX_TIME
+  getBoundedMaxTime(maxTime) {
+    let m = moment(maxTime, TIME_FORMAT);
+    if (!m.isValid()) {
+      return MAX_TIME;
+    }
+    return moment.min(MAX_TIME, m.subtract({minutes: 1}));
+  },
+
+  // Return the time 1 minute after specified minTime, bounded by MIN_TIME
+  getBoundedMinTime(minTime) {
+    let m = moment(minTime, TIME_FORMAT);
+    if (!m.isValid()) {
+      return MIN_TIME;
+    }
+    return moment.max(MIN_TIME, m.add({minutes: 1}));
+  },
+
   render() {
-    var style = {};
-    var help = {};
+    let style = {};
+    let help = {};
+    let date = moment(this.props.session.date, DATE_FORMAT);
     if (this.props.shouldValidate) {
       if (!this.props.session.date) {
         style.date = "error";
         help.date = "Required.";
-      } else if (!moment(this.props.session.date, 'MM/DD/YY').isValid()) {
+      } else if (!date.isValid()) {
         style.date = "error";
-        help.date = "Must be a valid date in MM/DD/YY format.";
-      }
-      if (!this.props.session.startTime) {
-        style.startTime = "error";
-        help.startTime = "Required.";
-      }
-      if (!this.props.session.endTime) {
-        style.endTime = "error";
-        help.endTime = "Required.";
-      } else if (this.props.session.endTime <= this.props.session.startTime) {
-        style.endTime = "error";
-        help.endTime = "Must end after it starts.";
+        help.date = `Must be a valid date in ${DATE_FORMAT} format.`;
       }
     }
 
+    // Start and end times have default values and should always be validated.
+    let startTime = moment(this.props.session.startTime, TIME_FORMAT);
+    let endTime = moment(this.props.session.endTime, TIME_FORMAT);
+    if (!this.props.session.startTime) {
+      style.startTime = "error";
+      help.startTime = "Required.";
+    } else if (!startTime.isValid()) {
+      style.startTime = "error";
+      help.startTime = "Invalid format.";
+    }
+    if (!this.props.session.endTime) {
+      style.endTime = "error";
+      help.endTime = "Required.";
+    } else if (!endTime.isValid()) {
+      style.endTime = "error";
+      help.endTime = "Invalid format.";
+    }
+    if (startTime.isValid() && endTime.isValid() && !endTime.isAfter(startTime)) {
+      style.endTime = "error";
+      help.endTime = "Must end after it starts.";
+    }
```

Comment:
Correct. Everything workshop-related is US-only, and not localized at the moment.